Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -574,10 +571,52 @@ void DrawFrame() {
         }
 
-        void GetHitTestResults(std::vector<HitResult>&, xr::Ray) const {
-            // TODO
+        void GetHitTestResults(std::vector<HitResult>& filteredResults, xr::Ray offsetRay) const {
+            @autoreleasepool {
+                if (session != nil && session.currentFrame != nil && session.currentFrame.camera != nil && [session.currentFrame.camera trackingState] == ARTrackingStateNormal) {
+                    if (@available(iOS 13.0, *)) {
+                        // Push the camera origin into a simd_float3.
+                        auto cameraOrigin = simd_make_float3(
+                                                             ActiveFrameViews[0].Space.Pose.Position.X,
+                                                             ActiveFrameViews[0].Space.Pose.Position.Y,
+                                                             ActiveFrameViews[0].Space.Pose.Position.Z);
+                        
+                        // Push the camera direction into a simd_quaternion.
+                        auto cameraDirection = simd_quaternion(
+                                                               ActiveFrameViews[0].Space.Pose.Orientation.X,
+                                                               ActiveFrameViews[0].Space.Pose.Orientation.Y,
+                                                               ActiveFrameViews[0].Space.Pose.Orientation.Z,
+                                                               ActiveFrameViews[0].Space.Pose.Orientation.W);
+                        
+                        // Load the offset ray and direction into simd equivalents.
+                        auto offsetOrigin = simd_make_float3(offsetRay.Origin.X, offsetRay.Origin.Y, offsetRay.Origin.Z);
+                        auto offsetDirection = simd_make_float3(offsetRay.Direction.X, offsetRay.Direction.Y, offsetRay.Direction.Z);
+
+                        // Construct the ARRaycast query compositing the camera and offset ray origin + direction targetting estimated planes.
+                        auto raycastQuery = [[ARRaycastQuery alloc]
+                                             initWithOrigin:(cameraOrigin + offsetOrigin)
+                                             direction:simd_act(cameraDirection, offsetDirection)
+                                             allowingTarget:ARRaycastTargetEstimatedPlane
+                                             alignment:ARRaycastTargetAlignmentAny];
+                        
+                        // Perform the actual raycast.
+                        auto rayCastResults = [session raycast:raycastQuery];
+                        
+                        // Process the results and push them into the results list.
+                        for (ARRaycastResult* result in rayCastResults) {
+                            filteredResults.push_back(transformToHitResult(result.worldTransform));
+                        }
+                    } else {
```

Comment:
If `offsetRay`'s origin is where the camera is (raycasting from the camera), then couldn't we hit test to a correct (according to the Ray) screen coordinate?

Reply:
Oh I see now you commented in the PR saying we could do this later.

Reply:
Yeah I'll do this later, currently we don't actually pass the offset ray in so effectively we're always hit testing from the center point